src/Authentication/Authentication/custom/common/Permissions.ps1

```diff
@@ -2,4 +2,82 @@
 #  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
 # ------------------------------------------------------------------------------
 
-$Permissions_MsGraphServicePrincipal = $null
+$permissionsMsGraphServicePrincipal = $null
+
+function GetPermissionsData {
+
+    # 2. Making a REST request to MS Graph
+    if ($null -eq $script:permissionsMsGraphServicePrincipal){
+        $script:permissionsMsGraphServicePrincipal = try {
+
+            Write-Host "Getting data from web service"
+            Invoke-MgGraphRequest -method GET 'https://graph.microsoft.com/v1.0/servicePrincipals?filter=appId eq ''00000003-0000-0000-c000-000000000000''' | select-object -expandproperty value
+
+        } catch {
+
+            Write-Host "Getting data from local file"
+            Get-Content $PSScriptRoot/datasample.json | Out-String | ConvertFrom-Json
```

Comment:
No it would work universally cause I added the datasample.json file to the most recent commit so it builds with every other program. And I opted to using $PSScriptRoot to get the current root of the files since the json file is under the custom/common folder